oak_services/proto/http_encap.proto

```diff
@@ -20,8 +20,8 @@ package oak.encap;
 
 // Protocol buffer encoding representing an HTTP request.
 message HttpRequest {
-  // The path to which the request was sent.
-  string path = 1;
+  // The URI (including scheme and authority) to which the request was sent.
```

Comment:
Not sure if it is true. The line that prints the `request.uri` in `examples/http_server/module/src/lib.rs` only prints "/", no scheme or authority.

Reply:
Really? It was printing the entire URI when I tried on my machine, which is why I decided to change it. But also just looking at the code, it takes the URI from the incoming request, so I would expect that the entire URI gets propagated.

Reply:
I'll wait until you confirm this, if what you say is correct then my change will probably break your use case, so I need to look into it more closely to understand what's going on.

Reply:
That makes sense. I get only the path ("/") for requests that are send from the rust client. For requests from the browser and the bash client the entire URI is printed.

Reply:
Thanks for confirming. Just to clarify: which Rust client are you referring to? AFAICT if it performs an actual HTTP request, it should result in the full URI anyways

Reply:
This is the line that sends the request:
https://github.com/project-oak/oak/blob/79de881763b06e83a00e80aeba740978ebd0949e/examples/http_server/client/src/main.rs#L61

Reply:
I tried various combinations, but I still always see it logged as "https://localhost:8080" (which is what I would expect). Do you have steps that make it log "/" as the URL / path?

Reply:
(since the discrepancy is already in `main`, I will go ahead with this PR anyways, but please do let me know how to reproduce)

Reply:
I run the server using `./scripts/runner --logs run-examples --example-name=http_server --client-variant=none` (inside docker), then I run the rust client:
```
cd examples/http_server/client
cargo run
```

Reply:
Thanks, I managed to reproduce that way, that's odd! Maybe the hyper client strips the uri before sending the request? not sure what's going on.

Reply:
I this the issue is that the hyper server reconstructs the URI from client headers. The raw HTTP request does not have the full URI of the request as a single field. In HTTP2 there are multiple headers (`:scheme`, `:host`, ":path") that can be used to reconstrutct the URI on the server. In HTTP1 the method and path are sent as the first line of the request body. A client can optionally also add a host header, but this is not required. Perhaps the rust client is defaulting to HTTP1 and does not use HTTP2.

Reply:
Good point @conradgrobler , I didn't realise that was the case. @rbehjati could we make the HTTP server only use HTTP2 anyways? Not sure whether that is sufficient to solve this issue, but it is something we discussed we should do anyways.

Reply:
What I remember from earlier discussions (and comments on #1261) is that HTTP server pseudo-node is supposed to only support HTTP/1 and JSON. Do we now have a consensus that we should completely switch to HTTP/2?

Reply:
I don't see why not, anyways it's just a matter of disabling HTTP1 with https://docs.rs/hyper/0.13.7/hyper/server/struct.Builder.html#method.http2_only , everything else should stay the same, right? The Rust client, `curl` and Chrome should all still keep working (but let me know if that's not the case).

Reply:
It would be interesting if everything stays the same. The pain of using hyper would pay off.

Reply:
I could not find a way to make the rust client work. I have created this PR: https://github.com/project-oak/oak/pull/1481
The error is "failed to lookup address information: Name or service not known". It could be the same problem we were experiencing with TLS for gRPC that did not work without a domain name. Let me know if you can think of any solutions.